lib/shopping_cart.rb

```diff
@@ -0,0 +1,61 @@
+class ShoppingCart
+  attr_reader :name, :capacity, :products
+
+  def initialize(name, capacity)
+    @name = name
+    @capacity = capacity
+    @products = []
+    @details = Hash.new
```

Comment:
Be careful with this here -- in order for this details hash to get filled, you first need to call the details method. I think that for this reason, you should pull line 8 into your details method and take care of making the whole hash there.